server/protocols/webadmin/webadmin-data/src/main/java/org/apache/james/webadmin/routes/RegexMappingRoutes.java

```diff
@@ -0,0 +1,104 @@
+import org.apache.james.webadmin.utils.ErrorResponder;
+import org.apache.james.webadmin.utils.JsonExtractor;
+import org.eclipse.jetty.http.HttpStatus;
+
+import com.google.common.annotations.VisibleForTesting;
+
+import io.swagger.annotations.Api;
+import io.swagger.annotations.ApiOperation;
+import io.swagger.annotations.ApiResponse;
+import io.swagger.annotations.ApiResponses;
+import spark.HaltException;
+import spark.Request;
+import spark.Response;
+import spark.Service;
+
+@Api(tags = "Adding")
+@Path(RegexMappingRoutes.BASE_PATH)
+@Produces(Constants.JSON_CONTENT_TYPE)
+public class RegexMappingRoutes implements Routes {
+
+    static final String BASE_PATH = "/mappings/regex";
+
+    private final RecipientRewriteTable recipientRewriteTable;
+    private final JsonExtractor<RegexMappingDTO> jsonExtractor;
+
+    @Inject
+    @VisibleForTesting
+    RegexMappingRoutes(RecipientRewriteTable recipientRewriteTable) {
+        this.recipientRewriteTable = recipientRewriteTable;
+        this.jsonExtractor = new JsonExtractor<>(RegexMappingDTO.class);
+    }
+
+    @Override
+    public String getBasePath() {
+        return BASE_PATH;
+    }
+
+    @Override
+    public void define(Service service) {
+        service.post(getBasePath(), this::addRegexMapping);
+    }
+
+    @POST
+    @Path(BASE_PATH)
+    @ApiOperation(value = "adding address-regex mappings to RecipientRewriteTable")
+    @ApiResponses(value = {
+        @ApiResponse(code = HttpStatus.NO_CONTENT_204, message = "No body created"),
+        @ApiResponse(code = HttpStatus.BAD_REQUEST_400, message = "Invalid body")
+    })
+    public HaltException addRegexMapping(Request request, Response response) throws Exception {
+        RegexMappingDTO regexMappingDTO = jsonExtractor.parse(request.body());
+        MappingSource mappingSource = extractMappingSource(regexMappingDTO);
+
+        try {
+            recipientRewriteTable.addRegexMapping(mappingSource, extractRegex(regexMappingDTO));
+        } catch (RecipientRewriteTableException e) {
+            throw ErrorResponder.builder()
+                .statusCode(HttpStatus.BAD_REQUEST_400)
+                .type(ErrorResponder.ErrorType.INVALID_ARGUMENT)
+                .message(e.getMessage())
+                .haltError();
+        }
+        return halt(HttpStatus.NO_CONTENT_204);
+    }
+
+    private MappingSource extractMappingSource(RegexMappingDTO regexMappingDTO) {
+        try {
+            return MappingSource.parse(regexMappingDTO.getSource());
+        } catch (IllegalArgumentException e) {
+            throw ErrorResponder.builder()
+                .statusCode(HttpStatus.BAD_REQUEST_400)
+                .type(ErrorResponder.ErrorType.INVALID_ARGUMENT)
+                .message("Invalid `source` field.")
+                .haltError();
+        }
+    }
+
+    private String extractRegex(RegexMappingDTO regexMappingDTO) {
+        try {
+            return regexMappingDTO.getRegex();
```

Comment:
I do not see when this call can fail. Can you develop? 

Also you can run test coverage. You will see if this bit of code (error management) is used, and remove it if not.